hyperspy/io.py

```diff
@@ -334,15 +334,15 @@ def load(filenames=None,
             filenames = _escape_square_brackets(filenames)
 
         filenames = natsorted([f for f in glob.glob(filenames)
-                               if os.path.isfile(f)])
+                               if os.path.isfile(f) or os.path.isdir(f)])
```

Comment:
I guess a directory should be accepted only when using zarr format?
```suggestion
                               if os.path.isfile(f) or (os.path.isdir(f) and os.path.splitext(f)[1] == '.zspy')])
```

If so, there are similar `if` statement which needs to be changed.